data/pools/transactionPool.go

```diff
@@ -401,6 +452,10 @@ func (pool *TransactionPool) ingest(txgroup []transactions.SignedTxn, params poo
 		if err != nil {
 			return err
 		}
+
+		// since this is the first time the transaction was added to the transaction pool, it would
+		// be a good time now to figure the group's ID.
+		txgroup.GroupTransactionID = txgroup.Transactions.ID()
```

Comment:
We would need this for future changes that Nicholas is working on and would be using the txnsync as a basis. This would also give us the ability to make few memory optimizations later in the signature verification cache.